dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,544 @@
+                **tasklet.out_connectors
+            }
+        ])
+        return internal_state_str, internal_state_var
+
+    def unparse_tasklet(self, sdfg: SDFG, dfg: StateSubgraphView, state_id: int,
+                        node: nodes.Node, function_stream: CodeIOStream,
+                        callsite_stream: CodeIOStream):
+
+        # extract data
+        state = sdfg.nodes()[state_id]
+        tasklet = node
+
+        # construct variables paths
+        unique_name: str = "top_{}_{}_{}".format(sdfg.sdfg_id,
+                                                 sdfg.node_id(state),
+                                                 state.node_id(tasklet))
+
+        # generate system verilog module components
+        parameter_string: str = self.generate_rtl_parameters(sdfg.constants)
+        inputs, outputs = self.generate_rtl_inputs_outputs(sdfg, tasklet)
+
+        # create rtl code object (that is later written to file)
+        self.code_objects.append(
+            CodeObject(
+                name="{}".format(unique_name),
+                code=RTLCodeGen.RTL_HEADER.format(name=unique_name,
+                                                  parameters=parameter_string,
+                                                  inputs="\n".join(inputs),
+                                                  outputs="\n".join(outputs)) +
+                tasklet.code.code + RTLCodeGen.RTL_FOOTER,
+                language="sv",
+                target=RTLCodeGen,
+                title="rtl",
+                target_type="",
+                additional_compiler_kwargs="",
+                linkable=True,
+                environments=None))
+
+        # generate verilator simulation cpp code components
+        inputs, outputs = self.generate_cpp_inputs_outputs(tasklet)
+        vector_init = self.generate_cpp_vector_init(tasklet)
+        num_elements = self.generate_cpp_num_elements()
+        internal_state_str, internal_state_var = self.generate_cpp_internal_state(
+            tasklet)
+
+        # add header code to stream
+        sdfg.append_global_code(cpp_code=RTLCodeGen.CPP_HEADER_TEMPLATE.format(
+            name=unique_name))
+
+        # add main cpp code to stream
+        callsite_stream.write(contents=RTLCodeGen.CPP_MAIN_TEMPLATE.format(
+            name=unique_name,
+            inputs=inputs,
+            outputs=outputs,
+            num_elements=num_elements,
+            vector_init=vector_init,
+            internal_state_str=internal_state_str,
+            internal_state_var=internal_state_var),
+                              sdfg=sdfg,
+                              state_id=state_id,
+                              node_id=node)
+
+    CPP_HEADER_TEMPLATE = """\
+// generic includes
+#include <iostream>
+
+// verilator includes
+#include <verilated.h>
+
+// include model header, generated from verilating the sv design
+#include "V{name}.h"
+"""
+
+    CPP_MAIN_TEMPLATE = """\
+if(DEBUG){{
```

Comment:
This macro should have a safer name, like `DACE_VERILATOR_ENABLE_DEBUG`. Even better: turn it into a `constexpr bool kDebugVerilator = DACE_VERILATOR_ENABLE_DEBUG;`.

Reply:
Done as soon as this pull request has been merged: https://github.com/spcl/dace/pull/463